src/STM32Ethernet.cpp

```diff
@@ -1,11 +1,12 @@
 #include "STM32Ethernet.h"
 #include "Dhcp.h"
 
-int EthernetClass::begin(uint8_t *mac_address, unsigned long timeout, unsigned long responseTimeout)
+int EthernetClass::begin(unsigned long timeout, unsigned long responseTimeout)
 {
   static DhcpClass s_dhcp;
   _dhcp = &s_dhcp;
-
+  uint8_t mac_address[6];
```

Comment:
I think mac address could be a private members of Ethernet Class with accessors (get/set)
It should be init to default mac based on UID.
Anyway user should be able to change it before calling begin